packages/js/product-editor/src/components/variations-table/variation-actions-menus/variation-actions.tsx

```diff
@@ -0,0 +1,148 @@
+	onClose: () => void;
+	supportsMultipleSelection: boolean;
+} ) {
+	return (
+		<div
+			className={ classNames( {
+				'components-dropdown-menu__menu': supportsMultipleSelection,
+			} ) }
+		>
+			<MenuGroup
+				label={
+					supportsMultipleSelection
+						? undefined
+						: sprintf(
+								/** Translators: Variation ID */
+								__( 'Variation Id: %s', 'woocommerce' ),
+								( selection as ProductVariation ).id
+						  )
+				}
+			>
+				{ supportsMultipleSelection ? (
+					<>
+						<UpdateStockMenuItem
+							selection={ selection }
+							onChange={ onChange }
+							onClose={ onClose }
+						/>
+						<SetListPriceMenuItem
+							selection={ selection }
+							onChange={ onChange }
+							onClose={ onClose }
+						/>
+					</>
+				) : (
+					<MenuItem
+						href={ ( selection as ProductVariation ).permalink }
+						target="_blank"
+						rel="noreferrer"
+						onClick={ () => {
+							recordEvent( 'product_variations_preview', {
+								source: TRACKS_SOURCE,
+								variation_id: ( selection as ProductVariation )
+									.id,
+							} );
+						} }
+					>
+						{ __( 'Preview', 'woocommerce' ) }
+					</MenuItem>
+				) }
+
+				<ToggleVisibilityMenuItem
+					selection={ selection }
+					onChange={ onChange }
+					onClose={ onClose }
+				/>
+				<VariationQuickUpdateMenuItem.Slot
+					group={ 'top-level' }
+					supportsMultipleSelection={ supportsMultipleSelection }
+				/>
+			</MenuGroup>
+			<MenuGroup>
+				<PricingMenuItem
+					selection={ selection }
+					onChange={ onChange }
+					onClose={ onClose }
+					supportsMultipleSelection={ supportsMultipleSelection }
+				/>
+				<InventoryMenuItem
+					selection={ selection }
+					onChange={ onChange }
+					onClose={ onClose }
+					supportsMultipleSelection={ supportsMultipleSelection }
+				/>
+				<ShippingMenuItem
+					selection={ selection }
+					onChange={ onChange }
+					onClose={ onClose }
+					supportsMultipleSelection={ supportsMultipleSelection }
+				/>
+				{ window.wcAdminFeatures[ 'product-virtual-downloadable' ] && (
+					<DownloadsMenuItem
+						selection={ selection }
+						onChange={ onChange }
+						onClose={ onClose }
+						supportsMultipleSelection={ supportsMultipleSelection }
+					/>
+				) }
+				<VariationQuickUpdateMenuItem.Slot
+					group={ 'secondary' }
+					supportsMultipleSelection={ supportsMultipleSelection }
+				/>
+			</MenuGroup>
+			<MenuGroup>
+				<MenuItem
+					isDestructive
+					label={
+						! supportsMultipleSelection
+							? __( 'Delete variation', 'woocommerce' )
+							: undefined
+					}
+					variant="link"
+					onClick={ () => {
+						onDelete( selection );
+						onClose();
+					} }
+					className="woocommerce-product-variations__actions--delete"
+				>
+					{ __( 'Delete', 'woocommerce' ) }
+				</MenuItem>
+				<VariationQuickUpdateMenuItem.Slot
+					group={ 'tertiary' }
+					supportsMultipleSelection={ supportsMultipleSelection }
+				/>
```

Comment:
We need to pass through the `selection` and `onChange` so that fills can access that in their `onClick` handler and actually do something!

We should also pass  the`onClose` callbacks, as that will be needed too for extensions to close the menu when they have done their action!